src/Components/DurationLineChart.tsx

```diff
@@ -0,0 +1,263 @@
+
+  if (hours > 0) {
+    if (minutes < 10) {
+      if (seconds < 10) {
+        return `${hours}:0${minutes}:0${seconds}`
+      } else {
+        return `${hours}:0${minutes}:${seconds}`
+      }
+    } else {
+      if (seconds < 10) {
+        return `${hours}:${minutes}:0${seconds}`
+      } else {
+        return `${hours}:${minutes}:${seconds}`
+      }
+    }
+  } else {
+    if (seconds < 10) {
+      return `${minutes}:0${seconds}`
+    } else {
+      return `${minutes}:${seconds}`
+    }
+  }
+}
+
+function findDomain(data: number[]): { min: number; max: number } {
+  let min = data[0]
+  let max = data[0]
+
+  for (let i = 1, length = data.length; i < length; i++) {
+    let v: number = data[i]
+    min = v < min ? v : min
+    max = v > max ? v : max
+  }
+
+  return { min, max }
+}
+
+function getTimeTicks(min: number, max: number, numTicks: number): number[] {
+  if (numTicks <= 2) {
+    return [min, max]
+  }
+
+  const range = max - min
+  const step = Math.ceil(range / (numTicks - 2) / 5000) * 5000 //round up to nearest 5000ms (5 seconds)
+  const lower = Math.trunc(min / step) * step //get the lower boundry
+
+  let ticks = [lower] //add the lower boundry
+
+  //add the ticks
+  for (let i = 1; i <= numTicks - 1; i++) {
+    ticks.push(lower + step * i)
+  }
+
+  return ticks
+}
+
+type DurationLineChartProps = {
+  chartTitle: string
+  data: Attempt[]
+  xAxisLabel: string
+  yAxisLabel: string
+}
+
+const DurationLineChart = ({
+  chartTitle,
+  data,
+  xAxisLabel,
+  yAxisLabel
+}: DurationLineChartProps) => {
+  //if there's no data return nothing
+  if (!data || data.length <= 0) {
+    return <React.Fragment></React.Fragment>
+  }
+
+  //get the domain for the x-axis
+  const xDomain = findDomain(
+    data.map((item: Attempt) => {
+      return item.x
+    })
+  )
+
+  //get the domain for the y-axis
+  const yDomain = findDomain(
+    data.map((item: Attempt) => {
+      return item.y
+    })
+  )
+
+  //get the ticks for the y-axis
+  const yTicks = getTimeTicks(yDomain.min, yDomain.max, 6)
+
+  return (
+    <VictoryChart //container for the chart
+      containerComponent={
+        //used for the tooltip, and the vertical indicator line
+        <VictoryVoronoiContainer
+          //only use the x-axis to determine which tooltip shows
+          voronoiDimension="x"
+          //labels expects a string so we need to return one here instead of just using an empty function () => {}
+          labels={() => {
+            return ''
+          }}
+          //draws the vertical indicator line
+          labelComponent={<Cursor />}
+        />
+      }
+      //max height of the graph
+      height={200}
+      //use the material theme instead of the grayscale theme (may be able to remove this once we add custom styling)
+      theme={VictoryTheme.material}
+    >
+      <VictoryLabel //table title
+        //center label horizontally in the container
+        dx="50%"
+        //style the title
+        style={{
```

Comment:
I’m not sure where I land on inline styles or create-an-object-at-the-top styles. Might make this section easier to read/write?

Reply:
death to inline styles! also don't you have MU? why aren't you using with styles??

Reply:
When I go through and style this chart I plan on doing this differently.  Right now it's just to make the chart somewhat readable.